modules/data.atmosphere/R/downscale.met.R

```diff
@@ -0,0 +1,425 @@
+        temp_min <- rep(NA,reso_len)}
+      if (all(is.na(model$air_temperature_min)) == FALSE){
+        for (x in seq_along(model$air_temperature_min)){
+          four <- vector()
+          for (n in seq_along(1:div)){
+            lowday <- x*div-wd*sd_step+n
+            highday <- x*div+wd*sd_step+n
+            if (lowday < 0){lowday = x+n
+            highday = x+n+wd*sd_step+n}
+            if (highday > reso_len){
+              highday = reso_len
+              lowday = x*div-wd*sd_step+n}
+            four[n] <- rnorm(1, mean = model$air_temperature_min[x],
+                             sd = sd(upscale_flux$air_temperature_min[seq(from=lowday,to=highday,by=sd_step)]))}
+          temp_min <- append(temp_min,four)}
+      }
+      
+      # Precipitation_flux
+      # this takes the daily total of precipitation and uses that as a total possible amount of precip. 
+      # It randomly distributes the values of precipitation
+      rand_vect_cont <- function(N, M, sd = 1) {
+        vec <- rnorm(N, M/N, sd)
+        vec / sum(vec) * M
+      }
+      precip <- vector()
+      for (x in seq_along(model$precipitation_flux)){
+        lowday <- (x-wd)*div
+        highday <- (x+wd)*div
+        if (lowday < 0){lowday = 0}
+        if (highday > reso_len){highday = reso_len}
+        four <- vector()
+        four <- rand_vect_cont(div,model$precipitation_flux[x], 
+                               sd = sd(upscale_flux$precipitation_flux[lowday:highday]))
+        four[four<0] = 0
+        precip <- append(precip,four)
+      }
+      
+      # Specific Humidity
+      spechum <- vector()
+      for (x in seq_along(model$specific_humidity)){
+        lowday <- (x-wd)*div
+        highday <- (x+wd)*div
+        if (lowday < 0){lowday = 0}
+        if (highday > reso_len){highday = reso_len}
+        four <- vector()
+        for (n in seq_along(1:div)){
+          four[n] <- rnorm(1, mean = model$specific_humidity[x], 
+                          sd = sd(upscale_flux$specific_humidity[lowday:highday]))}
+        spechum <- append(spechum,four)
+      }
+      spechum[spechum < 0] = 0
+      
+      # Winds
+      east <- vector()
+      for (x in seq_along(model$eastward_wind)){
+        lowday <- (x-wd)*div
+        highday <- (x+wd)*div
+        if (lowday < 0){lowday = 0}
+        if (highday > reso_len){highday = reso_len}
+        four <- vector()
+        for (n in seq_along(1:div)){
+          four[n] <- rnorm(1, mean = model$eastward_wind[x], 
+                          sd = sd(upscale_flux$eastward_wind[lowday:highday]))}
+        east <- append(east,four)
+      }
+  
+      north <- vector()
+      for (x in seq_along(model$northward_wind)){
+        lowday <- (x-wd)*div
+        highday <- (x+wd)*div
+        if (lowday < 0){lowday = 0}
+        if (highday > reso_len){highday = reso_len}
+        four <- vector()
+        for (n in seq_along(1:div)){
+          four[n] <- rnorm(1, mean = model$northward_wind[x],
```

Comment:
Since `rnorm` will generate as many values as you tell it, seems very inefficient to be doing this in a loop, generating one by one?